MERLinSample/MERLinSample/EventsListeners/AppDelegateEventsListener.swift

```diff
@@ -17,13 +17,13 @@ class AppDelegateEventsListener: EventsListening {
         self.router = router
     }
     
-    func registerToEvents(for producer: AnyEventsProducer, events: EventsProxy<AppDelegateEvent>) -> Bool {
-        events[event: AppDelegateEvent.didFinishLaunching].toVoid()
+    func registerToEvents(for producer: AnyEventsProducer, events: Observable<AppDelegateEvent>) -> Bool {
```

Comment:
You don't use `events` here at all but it's still passed as a parameter. See my previous question.

Reply:
It's used as generic constraint. Can't avoid it. I cannot cast AnyEventsProducer to a specific EventsProducer because of the nature of protocols with associated types in swift. But I can cast a concrete type to another concrete type. That's why I need events here.

Reply:
beside the producer has only `anyEvents` which is an `Observable<EventProtocol>` this events here is the missing var that makes an `AnyEventsProducer` an `EventsProducer`. It totally belongs here.